from_cpython/Lib/distutils/unixccompiler.py

```diff
@@ -246,6 +246,7 @@ def find_library_file(self, dirs, lib, debug=0):
         shared_f = self.library_filename(lib, lib_type='shared')
         dylib_f = self.library_filename(lib, lib_type='dylib')
         static_f = self.library_filename(lib, lib_type='static')
+        non_pyston_shared_f = '.'.join(shared_f.split('.')[:-2] + ['so'])
```

Comment:
I would prefer if this would explicitly check if `shared_f` ends in `.pyston.so` and only than remove it.
+ comments which mention that this is a pyston change before every change please